### PR TITLE
switch bootJDK download scripts to use adoptium binaries

### PR DESCRIPTION
**Pull request by gdams**
This fixes the failing JDK tip builds

fixes: https://github.com/adoptium/temurin-build/issues/2704

I really want to split these horrible blocks of code into a shared function but will work on that in a follow-up PR

### CODE REVIEW

**Review by jerboaa — APPROVED** [review]
Looks fine to me.

File: .azure-devops/build/steps/macOS/before.yml [hunk 1]
```diff
@@ -38,12 +38,12 @@ steps:
 
   # download JDKs from AdoptOpenJDK
   - bash: |
-      wget -O "${{ parameters.dependenciesDir }}/openjdk8.tar.gz" "https://api.adoptopenjdk.net/v3/binary/latest/8/ga/mac/x64/jdk/hotspot/normal/adoptopenjdk?project=jdk"
+      wget -O "${{ parameters.dependenciesDir }}/openjdk8.tar.gz" "https://api.adoptium.net/v3/binary/latest/8/ga/mac/x64/jdk/hotspot/normal/adoptium?project=jdk"
       if [ "${BOOTJDK_VERSION}" == "15" ]
       then
-          wget -O "${{ parameters.dependenciesDir }}/openjdk.tar.gz" "https://api.adoptopenjdk.net/v3/binary/latest/${BOOTJDK_VERSION}/ea/mac/x64/jdk/hotspot/normal/adoptopenjdk?project=jdk"
+          wget -O "${{ parameters.dependenciesDir }}/openjdk.tar.gz" "https://api.adoptium.net/v3/binary/latest/${BOOTJDK_VERSION}/ea/mac/x64/jdk/hotspot/normal/adoptium?project=jdk"
```
Comment (karianna) [inline .azure-devops/build/steps/macOS/before.yml:44]: I don't think adoptium can sever up 15 binaries no?

**Review by karianna — CHANGES_REQUESTED** [review]
Should we be using GH setup-java action for this?  Am also worried that adoptium only pulls down subset of adoptopenjdk API

**Comment by gdams** [comment]
> Should we be using GH setup-java action for this? Am also worried that adoptium only pulls down subset of adoptopenjdk API

No one of the main reasons we don't is because we want to test the code that pulls down boot JDK's. I've added another check which will fall back to adoptopenjdk if required.

**Comment by gdams** [comment]
Test failures are unrelated to this change

**Review by karianna — APPROVED** [review]

**Comment by karianna** [comment]
I'm going to merge this as it'll help with testing any emergency PRs WRT to java 17

**Comment by sxa** [comment]
@karianna @gdams I can see the fallback to adoptopenjdk in the linux script here but not the others. Isn't this introducing a bit of a risk? Given that we have the forwarding from adoptopenjdk->adoptium I'm not clear on why we're pushing this through during a lockdown period.

**Comment by karianna** [comment]
> @karianna @gdams I can see the fallback to adoptopenjdk in the linux script here but not the others. Isn't this introducing a bit of a risk? Given that we have the forwarding from adoptopenjdk->adoptium I'm not clear on why we're pushing this through during a lockdown period.

17/tip builds were failing for GH Actions which meant we didn't have PR testing on any changes we might need to push through during release week, so this was worth getting in (it certainly couldn't make it worse).

**Pull request merged**